### PR TITLE
Update Windows auth doc to 2.0

### PR DESCRIPTION
**Pull request by scottaddie**
Fixes https://github.com/aspnet/Docs/issues/4530

[Internal Review Page](https://review.docs.microsoft.com/aspnet/core/security/authentication/windowsauth?branch=pr-en-us-4617)

### CODE REVIEW

File: aspnetcore/security/authentication/windowsauth.md [hunk 1]
```diff
@@ -1,76 +1,65 @@
 ---
-title: Configure Windows Authentication in ASP.NET Core
+title: Configure Windows authentication in ASP.NET Core
 author: ardalis
-description: How to configure Windows Authentication in ASP.NET Core
-keywords: ASP.NET Core,
+description: This article describes how to configure Windows authentication in ASP.NET Core, using IIS Express, IIS, HTTP.sys, and WebListener.
```
Comment (Tratcher) [inline aspnetcore/security/authentication/windowsauth.md]: Http.Sys and WebListener? The versioning gets weird here huh?

**Comment by scottaddie** [comment]
@Tratcher Do you think we should provide a code snippet demonstrating `WindowsIdentity.RunImpersonated`? If so, do you have such a snippet handy?

File: aspnetcore/security/authentication/windowsauth.md [hunk 2]
```diff
@@ -82,43 +71,34 @@ Disable Anonymous Authentication and enable Windows Authentication.
...
+[!code-csharp[](windowsauth/sample/Program1x.cs?highlight=6-11)]
+
+---
+
+## Work with Windows authentication
+
+If your app allows Windows authentication and anonymous access, you can use the `[Authorize]` and `[AllowAnonymous]` attributes. Apps disallowing anonymous access don't require `[Authorize]`. In this scenario, the app is treated as requiring authentication, and anonymous requests are rejected. If the IIS site is configured to disallow anonymous access, the `[AllowAnonymous]` attribute doesn't allow anonymous requests. The `[AllowAnonymous]` attribute overrides `[Authorize]` attribute usage within apps which allow anonymous access. See [Simple Authorization](xref:security/authorization/simple) for more details.
```
Comment (Tratcher) [inline aspnetcore/security/authentication/windowsauth.md]: If the IIS site (or Http.Sys or WebListener servers) are configured to disallow anonymous access then the request never reaches your application and the `[AllowAnonymous]` attribute doesn't apply.

Comment (Tratcher) [reply aspnetcore/security/authentication/windowsauth.md]: Can we separate this into two sections? The simple case is that the servers disallow anonymous and no attributes are needed or applicable. If the servers support auth and anonymous then you need the attributes and one additional piece of config:
`services.AddAuthentication(IISDefaults.AuthenticationScheme);` or HttpSysDefaults.AuthenticationScheme though they have the same value. This only applies to 2.0. `[Authorize]` needs this to challenge anonymous requests for windows auth.

**Comment by Tratcher** [comment]
Sure. There's a sample on MSDN: https://msdn.microsoft.com/en-us/library/dn906220(v=vs.110).aspx. You'll have to downcast to a WindowsIdentity to get the AccessToken.

File: aspnetcore/security/authentication/windowsauth.md (hunk 1 above)
Comment (scottaddie) [reply aspnetcore/security/authentication/windowsauth.md]: Until we get versioning in place for the aspnet/Docs repo, we're going to see weird things like this.

File: aspnetcore/security/authentication/windowsauth.md [hunk 3]
```diff
@@ -97,7 +97,37 @@ Although Kestrel doesn't support Windows authentication, you can use [WebListene
...
+
+If using HTTP.sys, add the following to the `ConfigureServices` method:
+
+```csharp
+// HttpSysDefaults requires the following import:
+// using Microsoft.AspNetCore.Server.HttpSys;
+services.AddAuthentication(HttpSysDefaults.AuthenticationScheme);
+```
+
+### Disallow anonymous access
```
Comment (Tratcher) [inline aspnetcore/security/authentication/windowsauth.md]: Do this scenario first, it's more common.

**Comment by scottaddie** [comment]
@Tratcher Added a simple `RunImpersonated` code snippet. Does this look okay?

File: aspnetcore/security/authentication/windowsauth.md [hunk 4]
```diff
@@ -133,4 +133,42 @@ services.AddAuthentication(HttpSysDefaults.AuthenticationScheme);
...
+    // code omitted for brevity
+    ...
+
+    const int LOGON32_PROVIDER_DEFAULT = 0;
+    // This parameter causes LogonUser to create a primary token. 
+    const int LOGON32_LOGON_INTERACTIVE = 2;
+
+    // Call LogonUser to obtain a handle to an access token. 
+    SafeAccessTokenHandle safeAccessTokenHandle;
+    bool returnValue = LogonUser(userName, domainName, password,
```
Comment (Tratcher) [inline aspnetcore/security/authentication/windowsauth.md]: Yikes, too deep...
Start with this and cull as needed:
```
app.Run(async (context) =>
{
try
{
                var user = (WindowsIdentity)context.User.Identity;

                await context.Response.WriteAsync($"User: {user.Name}<br>");
                await context.Response.WriteAsync($"State: {user.ImpersonationLevel}<br>");

#if NET461
                using (user.Impersonate())
#else
                WindowsIdentity.RunImpersonated(user.AccessToken, () =>
#endif
                {
                     var useri = WindowsIdentity.GetCurrent();
                     var text = $"User: {useri.Name}<br>State: {useri.ImpersonationLevel}<br>";

                     var bytes = System.Text.Encoding.UTF8.GetBytes(text);
                     context.Response.Body.Write(bytes, 0 , bytes.Length);
                }
#if !NET461
                );
#endif
}
catch (Exception e)
{
                await context.Response.WriteAsync(e.ToString());  
}
});
```

**Comment by scottaddie** [comment]
@Tratcher This is ready for one last look.

**Pull request merged**